src/validators/dataclass.rs

```diff
@@ -176,6 +178,8 @@ impl Validator for DataclassArgsValidator {
                     ($args:ident, $get_method:ident, $get_macro:ident, $slice_macro:ident) => {{
                         // go through fields getting the value from args or kwargs and validating it
                         for (index, field) in self.fields.iter().enumerate() {
+                            if (!field.init) { continue };
```

Comment:
@dmontagu,

What about the case of a default value? Don't we want to support something like:

```py
def test_dataclass_field_default_factory_with_init():
    @pydantic.dataclasses.dataclass
    class Model:
        x: int = dataclasses.field(default_factory=lambda: 3, init=False)

    m = Model()
    assert 'x' in Model.__pydantic_fields__
    assert m.x == 3
    assert RootModel[Model](m).model_dump() == {'x': 3}
```

Reply:
Good catch, I didn't even realize that was a thing. Yes we should probably do that. I think we can just copy the handling of defaults from lower in this file.